tracer/src/Datadog.Trace/Ci/Ipc/CircularChannel.cs

```diff
@@ -0,0 +1,114 @@
+// <copyright file="CircularChannel.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Ci.Ipc;
+
+internal partial class CircularChannel : IChannel
+{
+    private const int DefaultBufferSize = 65536;
+    private const int HeaderSize = 4;
+
+    private const int PollingInterval = 25;
+    private const int MutexTimeout = 5000;
+
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(CircularChannel));
+
+    private readonly MemoryMappedFile _mmf;
+    private readonly Mutex _mutex;
+    private readonly int _bufferSize;
+    private long _disposed;
+
+    private Writer? _writer;
+    private Receiver? _receiver;
+
+    public CircularChannel(string fileName)
+        : this(fileName, DefaultBufferSize)
+    {
+    }
+
+    public CircularChannel(string fileName, int bufferSize)
+    {
+        // Check if the file name is an absolute path, if not let's use a temporary directory
+        if (!Path.IsPathRooted(fileName))
+        {
+            if (FrameworkDescription.Instance.OSPlatform == OSPlatformName.Linux)
+            {
+                // Use /dev/shm to store the memory mapped file on Linux
+                fileName = Path.Combine("/dev/shm", Path.GetFileName(fileName));
+            }
+            else
+            {
+                var folder = Path.Combine(Path.GetTempPath(), "shm");
+                Directory.CreateDirectory(folder);
```

Comment:
```
CreateDirectory(String)
Creates all directories and subdirectories in the specified path unless they already exist.
```

Create directory handles the directory and subdirectories.

Reply:
Handle of the crash in [33789f7](https://github.com/DataDog/dd-trace-dotnet/pull/5537/commits/33789f782135e4434bd00ced95ae9bc53e312545)

Reply:
> Create directory handles the directory and subdirectories

Sorry, I meant you're currently in the `if (!Path.IsPathRooted(fileName))` branch. You _only_ call `CreateDirectory()` if you're in this branch. But I think you should move the directory creation _outside_ the `if`.

Otherwise, if someone passes `C:\some\nonexistent\path.mmap` to the file, I think it could crash as the intermediate folders don't exist?

Reply:
ok, got it, changed in [3e7129a](https://github.com/DataDog/dd-trace-dotnet/pull/5537/commits/3e7129a2474d478833b989f8cc8628a9ea29b7f6)